book/stages/README.md

```diff
@@ -0,0 +1,162 @@
+```rust
+/// Sealed Ethereum full block.
+#[derive(Debug, Clone, PartialEq, Eq, Default, RlpEncodable, RlpDecodable)]
+pub struct BlockLocked {
+    /// Locked block header.
+    pub header: SealedHeader,
+    /// Transactions with signatures.
+    pub body: Vec<TransactionSigned>,
+    /// Ommer/uncle headers
+    pub ommers: Vec<SealedHeader>,
+}
+```
+
+The new block is then pre-validated, checking that the ommers hash and transactions root in the block header are the same in the block body. Following a successful pre-validation, the `BodyStage` loops through each transaction in the `block.body`, adding the transaction to the database. This process is repeated for every downloaded block body, with the `BodyStage` returning `Ok(ExecOutput { stage_progress: highest_block, reached_tip: true, done })` signaling it successfully completed. 
+
+<br>
+
+## SendersStage
+
+Following a successful `BodyStage`, the `SenderStage` starts to execute. The `SenderStage` is responsible for recovering the transaction sender for each of the newly added transactions to the database. At the beginning of the execution function, all of the transactions are first retrieved from the database. Then the `SenderStage` goes through each transaction and recovers the signer from the transaction signature and hash. The transaction hash is derived by taking the Keccak 256-bit hash of the RLP encoded transaction bytes. This hash is then passed into the `recover_signer` function.
+
+[File: crates/primitives/src/transaction/signature.rs](https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/signature.rs#L72)
+```rust
+
+    /// Recover signature from hash.
+    pub(crate) fn recover_signer(&self, hash: H256) -> Option<Address> {
+        let mut sig: [u8; 65] = [0; 65];
+
+        self.r.to_big_endian(&mut sig[0..32]);
+        self.s.to_big_endian(&mut sig[32..64]);
+        sig[64] = self.odd_y_parity as u8;
+
+        secp256k1::recover(&sig, hash.as_fixed_bytes()).ok()
+    }
+```
+
+In an [ECDSA (Elliptic Curve Digital Signature Algorithm) signature](), the "r", "s", and "v" values are three pieces of data that are used to mathematically verify the authenticity of a digital signature. ECDSA is a widely used algorithm for generating and verifying digital signatures, and it is often used in cryptocurrencies like Ethereum.
+
+The "r" is the x-coordinate of a point on the elliptic curve that is calculated as part of the signature process. The "s" is the s-value that is calculated during the signature process. It is derived from the private key and the message being signed. Lastly, the "v" is the "recovery value" that is used to recover the public key from the signature, which is derived from the signature and the message that was signed. Together, the "r", "s", and "v" values make up an ECDSA signature, and they are used to verify the authenticity of the signed transaction.
```

Comment:
Is there a resource we could link to here that goes more deeply into depth on ECDSA signatures, if readers are interested? (doesn't need to be in this PR)

Reply:
maybe in a further version of the docs we can allude to EIP-155 changes here? idk, maybe too low level, don't have much preference